services/artifactory/artifactory.service.js

```diff
@@ -0,0 +1,211 @@
+      namedParams: {
+        group: 'org.jfrog.artifactory.client',
+        artifact: 'artifactory-java-client-services',
+      },
+      queryParams: {
+        repos: 'libs-release-local',
+        remote: '1',
+        version: '1.0-SNAPSHOT',
+        server: 'https://my.artifactory.server/artifactory',
+        token: 'OTEfj876SDF2345JKIB/H&+FFGH',
+        username: 'jerry',
+        password: 'seinfeld',
+      },
+      staticPreview: Artifactory.render('2.1.0'),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'artifactory' }
+
+  static render(version) {
+    return renderVersionBadge({ version })
+  }
+
+  static orElseEnvVar(possibleValue, envVarName) {
+    if (possibleValue !== '') {
+      return possibleValue
+    } else {
+      const envValue = process.env[envVarName]
+      if (envValue && envValue !== 'undefined') {
+        return envValue
+      } else {
+        return ''
+      }
+    }
+  }
+
+  static getAuthorizationHeaderValue(
+    possibleToken,
+    possibleUsername,
+    possiblePassword,
+  ) {
+    // check for token value first
+    const artToken = Artifactory.orElseEnvVar(
+      possibleToken,
+      'ARTIFACTORY_TOKEN',
+    )
+    if (artToken !== '') {
+      return `Bearer ${artToken}`
+    }
+
+    // if no token then check for username and password
+    const artUsername = Artifactory.orElseEnvVar(
+      possibleUsername,
+      'ARTIFACTORY_USERNAME',
+    )
+    if (artUsername !== '') {
+      const artPassword = Artifactory.orElseEnvVar(
+        possiblePassword,
+        'ARTIFACTORY_PASSWORD',
+      )
+      if (artUsername !== '') {
+        return `Basic ${btoa(`${artUsername}:${artPassword}`)}`
+      }
+    }
+
+    return ''
+  }
+
+  async fetchVersion({ requestParams }) {
+    // cache results so as not to overload
+    const result = await getCachedResource({
+      url: requestParams.artifactoryEndpoint,
+      ttl: ONE_HOUR,
+      json: false,
+      scraper: response => response,
+      options: requestParams.options,
+    })
+
+    //  Request path here does not use caching but does
+    //  allow us to catch specific errors and return
+    //  more specific responses: can we do something
+    //  similar when using the cachedResource function?
+    //
+    // const { res, buffer } = await this._request({
+    //  url: requestParams.artifactoryEndpoint,
+    //  options: requestParams.options,
+    //  httpErrors: { 404: 'Not Found', 401: 'Bad Credentials' },
+    // })
+
+    return result
+  }
+
+  async handle(
+    { group, artifact },
+    { server, token, username, password, repos, remote, version },
+  ) {
+    // Artifactory can return plain/text or json for this endpoint we're hitting so
+    // we need to effectively accept all as we're not sure what we're going to be returned
+    const defaultHeaders = { Accept: '*/*' }
+
+    // dump query params to debug console
+    console.debug(`***** artifactory found the following query properties *****
```

Comment:
Don't know if you all care about keeping this in but it was generally useful for me when debugging to see what was coming through as values for the query parameters. Let me know if you want me to rip it out.